Include/pymacro.h

```diff
@@ -23,6 +23,45 @@
 #  define static_assert _Static_assert
 #endif
 
+
+// _Py_ALIGN_AS: this compiler's spelling of `alignas` keyword,
+// We currently use alignas for free-threaded builds only; additional compat
+// checking would be great before we add it to the default build.
+// Standards/compiler support:
+// - `alignas` is a keyword in C23 and C++11.
+// - `_Alignas` is a keyword in C11
+// - GCC & clang has __attribute__((aligned))
+//   (use that for older standards in pedantic mode)
+// - MSVC has __declspec(align)
+// - `_Alignas` is common C compiler extension
+// Older compilers may name it differently; to allow compilation on such
+// unsupported platforms, we don't redefine _Py_ALIGN_AS if it's already
+// defined. Note that defining it wrong (including defining it to nothing) will
+// cause ABI incompatibilities.
+#ifdef Py_GIL_DISABLED
+#   ifndef _Py_ALIGN_AS
+#       ifdef __cplusplus
+#           if (__cplusplus < 201103L) \
+                  && (defined(__GNUC__) || defined(__clang__))
+#               define _Py_ALIGN_AS(V) __attribute__((aligned(V)))
+#           else
+#               define _Py_ALIGN_AS(V) alignas(V)
+#           endif
+#       elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 202311L
+#           define _Py_ALIGN_AS(V) alignas(V)
+#       elif defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
+#           define _Py_ALIGN_AS(V) _Alignas(V)
+#       elif (defined(__GNUC__) || defined(__clang__)) \
+              && defined(__STDC_VERSION__) && __STDC_VERSION__ < 201112L
```

Comment:
Is not this condition redundant? The case `defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L` is covered above, so the only special case for which this condition would be needed is when `__STDC_VERSION__` is not defined while `__GNUC__` or `__clang__` are defined. Is it even possible?

Reply:
Good catch, thanks!